Baboon_stuff/Baboon_Reader.cpp

```diff
@@ -1,77 +1,152 @@
-// reading a text file
 #include <iostream>
 #include <fstream>
 #include <string>
 #include <vector>
 #include <sstream>
 #include <algorithm>
 #include <map>
+#include <utility>      // std::pair, std::make_pair
+#include <bits/stdc++.h>
 using namespace std;
-/*store timestamp location long lat and individual*/
 
-main () {
-  //declaring variables
-  cout<<"hi";
-  string line;
-  ifstream myfile ("Baboon_data.csv");
-  vector<string> baboon_names;
-  map<string, vector<vector<string>>> baboon_personal_data;
 
-  if (myfile.is_open()) //opens file
-  {
-
-    while ( getline (myfile,line) ) //assigns each line of the file to the string line.
-    {
-
-      stringstream s_stream(line);// don't quite understand this yet. The documentation was a little confusing, will read further.
-
-      vector<string> result;
-
-      while(s_stream.good()) //goes through each line (while there are lines to go through) and seperates based on the delimeter (the comma)
+class Baboon_Reader {
+public:
+  map<string, vector<tuple<double,double,double>>> readFile(){

```

Comment:
You may want to consider the syntax for creating aliases:
```c++
using myname = std::map< std::string, std::vector< std::tuple< double, double, double > > >;
```
Then elsewhere in your code you can use the better named alias than this really long compound data type.

Reply:
This is very helpful, I implemented it this morning, makes the code much more attractive. Thank you